### PR TITLE
Add `scripts-dev/lint.sh`

### PR DESCRIPTION
**Pull request by reivilibre**

### CODE REVIEW

**Comment by callahad** [comment]
Can I ask what the motivation is for this PR?

**Comment by reivilibre** [comment]
We have this very useful script in Synapse which runs all the lints; as a matter of convenience and consistency, it seemed a good idea to have it here also.

**Comment by clokep** [comment]
I usually attempt to run this for sygnal and sydent and then cry and look at CI configs to see what I'm supposed to run. I'd love to have this! :)

File: scripts-dev/lint.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+#
+# Runs linting scripts over the local Synapse checkout
```
Comment (erikjohnston) [inline scripts-dev/lint.sh:3]:
```suggestion
# Runs linting scripts over the local Sygnal checkout
```

File: scripts-dev/lint.sh [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+else
+  # If we were not asked to lint changed files, and no paths were found as a result,
+  # then lint everything!
+  if [[ -z ${files+x} ]]; then
+    # Lint all source code files and directories
+    # Note: this list aims to mirror the one in tox.ini
+    files=(
+      "sygnal"
+      "tests"
+      "scripts-dev/old_deps.py"
```
Comment (erikjohnston) [inline scripts-dev/lint.sh:87]: Any reason not to do the whole directory?

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**